cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,6 +12,100 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 
+function retry {
+  local -r tries=$1
+  local -r wait_time=$2
+  local -r action=$3
+  local -r wait_message=$4
+
+  local result=$(eval $action)
+  for i in $(seq $tries); do
+    if [[ -z $result ]] ; then
+      echo "[$i/$tries] $wait_message"
+      sleep $wait_time
+      result=$(eval $action)
+    else
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+function wait_pod {
+  local namespace=$1
+  local label=$2
+
+  local -r tries=60
+  local -r wait_time=5
+
+  local -r wait_message="Waiting for pods with $label to create"
+  local -r error_message="Pods with  label $label at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  if [[ $label != "" ]];then
+    label="-l $label"
+  fi
+
+  local -r action="_kubectl get pod $namespace $label -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return  0
+  echo $error_message && return 1
+}
+
+function wait_k8s_object {
+  local -r object_type=$1
+  local -r name=$2
+  local namespace=$3
+
+  local -r tries=60
+  local -r wait_time=3
+
+  local -r wait_message="Waiting for $object_type $name"
+  local -r error_message="$object_type $name at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  local -r action="_kubectl get $object_type $name $namespace -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return 0
+  echo $error_message && return  1
+}
+
+
+function wait_for_taint_absence {
+  local -r taint=$1
+
+  local -r tries=60
+  local -r wait_time=5
+
+  local -r wait_message="Waiting for $taint taint absence"
+  local -r error_message="Taint $taint $name did not removed"
+  local -r action='[[ -z $(_kubectl get nodes -o custom-columns=taints:.spec.taints[*].effect --no-headers | grep -i $taint) ]] && echo false || echo '
```

Comment:
would you please inject a hiccup, for this command
```
 local -r action='[[ -z $(_kubectl get nodes -o custom-columns=taints:.spec.taints[*].effect --no-headers | grep -i $taint) ]] && echo false || echo '
````
its meant to be echo false ? or should be false ?
because the condition for retry is -z, and echo false print the string false
(unless there is a double eval that i miss)
just want to make sure please it actually retry on hiccups, unless you decide its not part of this PR.
thanks

Reply:
What hiccups? Why is it needed?
You should retry only if you can reason for it, and then I want to see it explained in the commit.

Reply:
based on real story (ask Petr)
once in a while a kubectl get command has timeout
and the logic right now, when it happens would thing that no taints exists
(no stdout of kubectl when the command fails | grep taint == nothing)

the code wont be a barrier for waiting for the taints to be removed,
unless it will do the kubectl again few times until it gets the actual state

see please the PR i mentioned that Petr did, also i saw it couple times happening both on prow and locally
again i wont block, but its a true error from what i understand so far